src/main/java/org/junitpioneer/jupiter/ByteRangeSource.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+/**
+ * {@code @ByteRangeSource} is an {@link ArgumentsSource} which provides access to a range of {@code byte} values.
+ *
+ * <p>The supplied values will be provided as arguments to the annotated {@code @ParameterizedTest} method.
+ *
+ * @since 0.5
+ * @see ArgumentsSource
+ * @see org.junit.jupiter.params.ParameterizedTest
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@ArgumentsSource(RangeSourceProvider.class)
+public @interface ByteRangeSource {
+
+	/**
+	 * The starting point of the range, inclusive.
+	 */
+	byte from();
+
+	/**
+	 * The end point of the range, exclusive.
+	 */
+	byte to();
+
+	/**
+	 * The size of the step between the {@code from} and the {@code to}.
+	 */
+	byte step() default 1;
+
+	/**
+	 * Whether the range is closed (inclusive of the {@link #to()}) or not.
+	 */
+	boolean closed() default false;
```

Comment:
A totally non-representative poll of three guys in Twitch chat (plus me!) showed that we all prefer `true` as a default here. It's odd in Java terms, but if I write `@RangeSource(from = 1, to = 3)`, I expect `3` to be in there. Your take on this?

Reply:
Looking at other similar APIs, like `IntStream.range` or even `List.subList` or `String.substring`, the "to" argument is always exclusive. I think having "to" here inclusive by default would be confusing than having it exclusive by default.

Reply:
Yeah, it's "Java conventions" vs "common sense". Let's leave it as it is and see whether users complain. We're pre-1.0, so we can break all the things! :wink: